### PR TITLE
Add blog support

### PR DESCRIPTION
**Pull request by menxs**
- [X] I acknowledge my contribution to the website does not assert comparative or superlative differences of one product, project, company or individual over another.

# Screenshots
## Create post
Create
## Edit post
Edit
## Show post
Show
## Show archived posts
Show archived

### CODE REVIEW

File: lib/erlef_web/controllers/blog_controller.ex [hunk 1]
```diff
@@ -83,7 +218,51 @@ defmodule ErlefWeb.BlogController do
...
+
+  defp categories(conn) do
+    Blog.categories_allowed_to_post(conn.assigns.current_user)
+  end
+
+  defp allowed_to_edit(conn, category) do
+    if category in categories(conn) do
+      :ok
+    else
+      :error
```
[the remark this answers is not in the document]
Comment (starbelly) [reply lib/erlef_web/controllers/blog_controller.ex]: > To get an error such that you're not authorized to edit something, either you manually tried to put the url of edition or archived post, put a category that you dont have in the post form.

That's fair and actually now that I think about it that really should be caught higher up the stack anyway. 

> Now that i write this, if your session dies and you click to edit a post you'll be redirected to not_found, in wich cases we should give the feedback that you're not authorized for that action? Thoughts?

What should currently happen is you get redirected to / with an error message stating you need to be logged in to perform an action.

File: lib/erlef_web/views/view_helpers.ex [hunk 2]
```diff
@@ -98,6 +98,11 @@ defmodule ErlefWeb.ViewHelpers do
     !!assigns[:current_user]
   end
 
+  def allowed_to_post?(assigns) do
+    true
```
Comment (starbelly) [inline lib/erlef_web/views/view_helpers.ex:102]: Looks like `true` should be removed here.

File: priv/posts/marketing/20210714233950_getting-to-know-paulo-oliveira.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Getting to know to Paulo F. Oliveira
 ---
 
 
-<img src="/images/getting-to-know/poliveira.png" class="img-fluid" alt="Paulo F. Oliveira"/>
+<img src="/images/getting-to-know/poliveira.png" width="100%" height="auto" alt="Paulo F. Oliveira"/>
```
Comment (starbelly) [inline priv/posts/marketing/20210714233950_getting-to-know-paulo-oliveira.md:14]: Why these changes?

File: priv/posts/marketing/20210331201645_getting-to-know-us-prodriguez.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ Getting to know Pehuén Rodriguez
 ---
 
 
-<img src="/images/getting-to-know/prodriguez.png" class="img-fluid" alt="Pehuén Rodriguez"/>
+<img src="/images/getting-to-know/prodriguez.png" width="100%" height="auto" alt="Pehuén Rodriguez"/>
```
Comment (starbelly) [inline priv/posts/marketing/20210331201645_getting-to-know-us-prodriguez.md:14]: Why these changes?

File: priv/repo/migrations/20210924114509_add_post_history_on_posts.exs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+defmodule Erlef.Repo.Migrations.AddPostHistoryOnPosts do
```
Comment (starbelly) [inline priv/repo/migrations/20210924114509_add_post_history_on_posts.exs:1]: We should just collapse this into the main migration since it's part of the same PR.

File: lib/erlef_web/templates/blog/form.html.eex [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        <%= label f, :title %>
+        <%= text_input f, :title, class: "form-control form-control-lg", oninput: "save(this)"%>
+        <%= error_tag f, :title %>
+        <%= error_tag f, :slug %>
+      </div>
+
+      <div class="form-group">
+        <%= label f, :authors %>
+        <div class="input-group flex-row">
+          <%= text_input f, :authors, class: "post-form-authors", value: authors_value(f), onchange: "save(this)"%>
```
Comment (starbelly) [inline lib/erlef_web/templates/blog/form.html.eex:20]: We might want to add an outline / border to this field so it's more clear that it's a field, especially in regards to error messages, what do you think?

File: priv/posts/marketing/20210714233950_getting-to-know-paulo-oliveira.md (hunk 3 above)
Comment (menxs) [reply priv/posts/marketing/20210714233950_getting-to-know-paulo-oliveira.md:14]: The Markdown sanitizer removes the class="img-fluid"  from the img tag, so the images cut out, width="100%" height="auto" fixes it.

File: priv/posts/marketing/20210331201645_getting-to-know-us-prodriguez.md (hunk 4 above)
Comment (menxs) [reply priv/posts/marketing/20210331201645_getting-to-know-us-prodriguez.md:14]: Same as above, Markdown sanitizer removes the class attribute

File: lib/erlef_web/views/view_helpers.ex (hunk 2 above)
Comment (menxs) [reply lib/erlef_web/views/view_helpers.ex:102]: Great catch!

File: priv/posts/marketing/20210331201645_getting-to-know-us-prodriguez.md (hunk 4 above)
Comment (starbelly) [reply priv/posts/marketing/20210331201645_getting-to-know-us-prodriguez.md:14]: Oh... right, you mentioned that and I forgot it... excellent attention to detail on your part!

**Comment by menxs** [comment]
I squashed the commits as you asked me, but I tried to keep the changes grouped in a few of them. If you want to, I can squash the commits further :)
